lib/logstash/inputs/http.rb

```diff
@@ -8,7 +8,15 @@
 require "base64"
 require "rack"
 
-class Puma::Server
+
+##
+# We keep the redefined method in a new http server class, this is because
+# in other parts of logstash we might be using puma as webserver, for example
+# in the sinatra part we need thie method to actually return the REQUEST_PATH, 
+# so it can actually infer the right resource to use.
+# Fixes https://github.com/logstash-plugins/logstash-input-http/issues/51
+##
+class HTTPPumaServer < Puma::Server
```

Comment:
At some point we can consolidate the 2 http servers and we can delegate all requests that come to this input to go to main LS http server